src/lib/db.mjs

```diff
@@ -51,18 +57,34 @@ export async function makeFaunaRequest({
 
   const fetchArgs = {
     method,
-    headers: { AUTHORIZATION: `Bearer ${argv.secret}` },
+    headers: { AUTHORIZATION: `Bearer ${secret}` },
   };
 
   if (body) fetchArgs.body = body;
 
   const response = await fetch(fullUrl, fetchArgs);
-
   const obj = await response.json();
 
   if (obj.error && shouldThrow) {
-    throw new Error(obj.error.message);
+    const err = new CommandError(obj.error.message);
+    err.name = obj.error.code;
+
+    if (obj.error.code === "unauthorized") {
+      err.message = "The database secret provided is invalid.";
+    }
+
+    throw err;
   }
 
   return obj;
 }
+
+/**
+ * @param {fetchParameters} opts
+ * @returns {Promise<Object>}
+ */
+export function makeRetryableFaunaRequest(opts) {
+  return retryInvalidCredsOnce(opts.secret, (secret) =>
+    makeFaunaRequest({ ...opts, secret }),
+  );
+}
```

Comment:
not for this PR, but i think it'd be lower mental overhead to have just one entrypoint (`makeFaunaRequest`) and pass it `opts, argv`, and have it choose if it should be retryable or not. exporting multiple entrypoints and expecting consumers to pick correctly is more brittle.